org.osgi.util.converter/src/org/osgi/util/converter/ConvertingImpl.java

```diff
@@ -216,6 +220,14 @@ else if (rt instanceof ParameterizedType) {
 			return convertToCollectionType(converter);
 		} else if (isMapType(targetAsClass, targetAsJavaBean, targetAsDTO)) {
 			return (T) convertToMapType(converter);
+		} else if (Optional.class.equals(targetAsClass)) {
```

Comment:
I think that these checks are in the wrong place. This set of if blocks is checking to see whether the target type is non-scalar. I think that all of the checks for optional should be in `trySpecialCases` to match other specially handled types. See https://github.com/osgi/osgi/blob/6b2c36000810b9b696ee6a26e709e8b8b832c830/org.osgi.util.converter/src/org/osgi/util/converter/ConvertingImpl.java#L1002